Source/WebCore/editing/Editor.cpp

```diff
@@ -986,7 +986,14 @@ RefPtr<Element> Editor::findEventTargetFrom(const VisibleSelection& selection) c
 
 RefPtr<Element> Editor::findEventTargetFromSelection() const
 {
-    return findEventTargetFrom(document().selection().selection());
+    // https://www.w3.org/TR/clipboard-apis/#fire-a-clipboard-event says:
+    // If the context is editable, then
+    // Set target to be the element that contains the start of the visible selection or cursor in document order, or the body element if there is no visible selection or cursor.
+    if (isEditablePosition(document().selection().selection().start()))
```

Comment:
I thinks we need to do this whenever selection is a range. Not just when it's editable.